docs/shipping/Code/java.md

```diff
@@ -494,6 +494,235 @@ If the log appender does not ship logs, add `<inMemoryQueue>true</inMemoryQueue>
+            </manifest>
+          </archive>
+        </configuration>
+        <executions>
+          <execution>
+            <id>make-assembly</id>
+            <phase>package</phase>
+            <goals>
+              <goal>single</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+
+```
+
+
+
+#### Add the OpenTelemetry controller
+
+```java
+package com.logzio.otel;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.logs.GlobalLoggerProvider;
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import io.opentelemetry.api.common.Attributes;
+
+public class OpenTelemetryConfig {
+
+    private static final String DEFAULT_ENDPOINT = "https://otlp-listener.logz.io/v1/logs";
+    private static final String LOGZ_IO_TOKEN = "<LOG-SHIPPING-TOKEN>";
+    private static final String SERVICE_NAME = "java-otlp";
+
+    public void initializeOpenTelemetry() {
+
+        // set service name on all OTel signals
+        Resource resource = Resource.getDefault().merge(Resource.create(
+                Attributes.of(ResourceAttributes.SERVICE_NAME, SERVICE_NAME)));
+
+        // Set up the OTLP log exporter with the endpoint and necessary headers
+        OtlpHttpLogRecordExporter logExporter = OtlpHttpLogRecordExporter.builder()
+                .setEndpoint(DEFAULT_ENDPOINT)
+                .addHeader("Authorization", "Bearer " + LOGZ_IO_TOKEN)
+                .build();
+
+        // Initialize the logger provider
+        SdkLoggerProvider sdkLoggerProvider = SdkLoggerProvider.builder()
+                .setResource(resource)
+                .addLogRecordProcessor(BatchLogRecordProcessor.builder(logExporter).build())
+                .build();
+
+        // create sdk object and set it as global
+        OpenTelemetrySdk sdk = OpenTelemetrySdk.builder()
+                        .setLoggerProvider(sdkLoggerProvider)
+                        .build();
+        GlobalOpenTelemetry.set(sdk);
+        // connect logger
+        GlobalLoggerProvider.set(sdk.getSdkLoggerProvider());
+        // Add hook to close SDK, which flushes logs
+        Runtime.getRuntime().addShutdownHook(new Thread(sdk::close));
+    }
+}
+```
+
+#### Run the application
+
+   ```bash
+   mvn clean package
+   java -jar target/*.jar
+   ```
+
+
+### Check Logz.io for your logs
+
+
+Allow some time for data ingestion, then open [Open Search Dashboards](https://app.logz.io/#/dashboard/osd).
+
+Encounter an issue? See our [log shipping troubleshooting](https://docs.logz.io/docs/user-guide/log-management/troubleshooting/log-shipping-troubleshooting/) guide.
+
+
+ 
```

Comment:
Add the code for logback.xml as well. 
See the link to it here:
https://github.com/logzio/opentelemetry-examples/blob/main/java/logs/src/main/resources/logback.xml